### PR TITLE
Document diff appearance and Git coloring

### PR DESCRIPTION
**Pull request by gerhardol**
Basically what was implemented in https://github.com/gitextensions/gitextensions/pull/11677
and https://github.com/gitextensions/gitextensions/pull/11590

See build in https://git-extensions-documentation.readthedocs.io/en/tmp-4.3/settings.html#diff-appearance
(with a few additional changes).

@pmiossec suggested the UI was updated, it may be a followup. I prefer to merge this PR first though...

### CODE REVIEW

File: source/settings.rst [hunk 1]
```diff
@@ -1095,7 +1097,91 @@ This page contains settings for the Git Extensions :ref:`commit` dialog. Note th
...
+    The patch view shows the changes in a file as a patch. This is the default view.
+
+    The appearance can to some extent be controlled by Git Settings, see also :ref:`settings-diff-coloring`.
+
+  .. setting:: Git word diff
+
+    The appearance uses Git word diff as described in `GitDiff#--word-diff <https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---word-diffltmodegt>`_
+
+    This is not a "patch" view, and so cannot be used to apply line patches.
+    It is intended as a contrast to the default view why Git default view is changed to show a minimal diff.
```
Comment (pmiossec) [inline source/settings.rst:1118]: I think that this sentence should be rewritten...

**Comment by pmiossec** [comment]
Have a look at the changes I have pushed. I think that fixes some issues...

**Comment by gerhardol** [comment]
> Have a look at the changes I have pushed. I think that fixes some issues...

Thanks. I aligned the links to Git documentation some more, but did not add the anchors to the display test for the others.
Instead of your GitDiff display text I changed to the existing use of "git diff", could be git-diff too (as in the url).

File: source/settings.rst (hunk 1 above)
Comment (pmiossec) [reply source/settings.rst]: Read it again and again and I still don't get the sense of this sentence 😕

Is `It is intended, as a contrast to the default view, to show a minimal diff.` not good enough?

Comment (gerhardol) [reply source/settings.rst]: Yes, that is better, will change.

**Comment by gerhardol** [comment]
Reminder about this pr that describes some changes in master.
@gitextensions/git-extensions-source

**Review by mstv — APPROVED** [review]
My 2ct: 👍

**Comment by gerhardol** [comment]
Cleaned up the commits, the latest updates had been pushed to #151 (separately approved)

**Review by RussKie — APPROVED** [review]

**Pull request merged**